cfme/infrastructure/virtual_machines.py

```diff
@@ -954,22 +1074,24 @@ def get_number_of_vms(do_not_navigate=False):
 @navigator.register(Template, 'All')
 @navigator.register(Vm, 'All')
 class VmAllWithTemplates(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+    VIEW = VmsTemplatesAllView
+    prerequisite = NavigateToObject(Server, 'LoggedIn')
 
-    def step(self, *args, **kwargs):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', '/vm_infra/explorer')
-        accordion.tree('VMs & Templates', 'All VMs & Templates')
+    def step(self):
+        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Virtual Machines')
+        self.view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
 
     def resetter(self, *args, **kwargs):
-        reset_page()
+        self.view.reset_page()
 
-    def am_i_here(self, *args, **kwargs):
-        return match_page(summary='All VMs & Templates')
+    def am_i_here(self):
+        return self.view.is_displayed and self.view.sidebar.vmstemplates.is_opened
```

Comment:
lkhomenk CFMENavigateStep.am_i_here calls self.view.is_displayed by default. These checks should be moved into the view's is_displayed method.